src/Generator/PropertyGenerator.php

```diff
@@ -304,4 +343,14 @@ public function omitDefaultValue(bool $omit = true)
 
         return $this;
     }
+
+    public function getType(): ?TypeGenerator
+    {
+        return $this->type;
+    }
+
+    public function setType(TypeGenerator $type): void
```

Comment:
Can the type be somehow reset? If we allow `?TypeGenerator $type` here, then, the constructor can call this method all the time (and a conditional disappears too)

Reply:
I simply followed convention like all other properties of PropertyGenerator.

I couldn't think of a need to RESET the $type, that's why I took the '?' out of `?TypeGenerator $type`

Reply:
> I couldn't think of a need to RESET the `$type`

This class is mostly a builder, so I can imagine `$p = PropertyGenerator::fromReflection(...); $p->setType(null); ...`

Reply:
49d33430a8d09699aa5aed85bcb0b3bfdf7033c1